core/devdoc/oop_module_requirements.md

```diff
@@ -0,0 +1,106 @@
+Module Requirements
+===================
+
+## Overview
+
+This is the documentation for an out-of-process module that communicates with
+other modules through the Azure IoT Field Gateway. Out-of-process modules need
+to implement the well defined Module Interface. Each out-of-process module will
+additionally need to attach and detach from the gateway and will either need to
+manage the worker thread manually or use the convenience methods supplied to
+start and halt the worker thread.
+
+## References
+
+[Azure IoT Field Gateway Module Interface](module.md)
+
+![High Level Design Diagrams](media/oop_module_hld.PNG)
+
+## Exposed Types
+
+```c
+typedef OOP_MODULE * OOP_MODULE_HANDLE;
+```
+
+## Exposed API
+
+### OopModule_Attach
+
+`OopModule_Attach` will provide the OopModule library with the caller's
+"Module API" and the connection key required to communicate with the Azure IoT
+Field Gateway. When `OopModule_Attach` is called the OopModule library will begin
```

Comment:
We used to call it that, pre-GA. Then "Field" got dropped. So it wasn't all in your head...